### PR TITLE
Bug 1678472: Create external name service default/openshift to resolve 'openshift.default.svc' to 'kubernetes.default.svc'

### PR DESCRIPTION
**Pull request by pravisankar**
- This will maintain backward compatibility with openshift 3.x

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge PTAL

**Comment by pravisankar** [comment]
@openshift/sig-network-edge ready for review, PTAL

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -268,6 +274,62 @@ func (h *Handler) ensureCoreDNSForClusterDNS(dns *dnsv1alpha1.ClusterDNS) error
...
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "openshift",
+			Namespace: "default",
+		},
+		Spec: corev1.ServiceSpec{
+			Type:         corev1.ServiceTypeExternalName,
+			ExternalName: "kubernetes.default.svc.cluster.local",
+			Ports: []corev1.ServicePort{{
```
Comment (ironcladlou) [inline pkg/stub/handler.go]: Is `Ports` actually required here? I don't have a cluster up yet to test, but if memory serves when I tested this solution only `Type` and `ExternalName` were required to get the CNAME.

Comment (pravisankar) [reply pkg/stub/handler.go]: Did not find enough documentation on ExternalName service type. Tested accessing openshift service on port 443 without adding port entries in the openshift service spec and it is working as expected. So, ports field is not required here.

**Comment by pravisankar** [comment]
@ironcladlou updated, PTAL

**Comment by pravisankar** [comment]
```
msg="4 errors occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.route_net[2]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.2: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route_table_association.route_net[0]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.0: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route.to_nat_gw[4]: 1 error occurred:"
level=error msg="\t* aws_route.to_nat_gw.4: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route.to_nat_gw[3]: 1 error occurred:"
level=error msg="\t* aws_route.to_nat_gw.3: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
```
/retest

**Comment by ironcladlou** [comment]
I'm glad this was relatively straightforward to solve — thanks!

/lgtm

**Pull request merged**